control-announcements/src/main/java/org/oskari/announcements/actions/AnnouncementsServiceMybatisImpl.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    public JSONObject getAdminAnnouncements() {
+        LOG.debug("get admin announcements");
+        final SqlSession session = factory.openSession();
+        JSONObject data = new JSONObject();
+        try {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            data = parser.parseAnnouncementsMap(mapper.getAdminAnnouncements());
+            session.commit();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to get admin announcements", e);
+        } finally {
+            session.close();
+        }
+        return data;
+    }
+
+    public JSONObject getAnnouncements() {
+        LOG.debug("get announcements");
+        final SqlSession session = factory.openSession();
+        JSONObject data = new JSONObject();
+        try {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            data = parser.parseAnnouncementsMap(mapper.getAnnouncements());
+            session.commit();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to get announcements", e);
+        } finally {
+            session.close();
+        }
+        return data;
+    }
+
+    public int updateAnnouncement(final Announcement announcement) {
+        LOG.debug("update announcement");
+        final SqlSession session = factory.openSession();
+        int updateId = -1;
+        try {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            updateId = mapper.updateAnnouncement(announcement);
+            session.commit();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to update announcement", e);
+        } finally {
+            session.close();
+        }
+        return updateId;
+    }
+
+    public int saveAnnouncement(final Announcement announcement) {
+        LOG.debug("save new announcement");
+        final SqlSession session = factory.openSession();
+        int saveId = -1;
+        try {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            saveId = mapper.saveAnnouncement(announcement);
+            session.commit();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to save announcement", e);
+        } finally {
+            session.close();
+        }
+        return saveId;
+    }
+
+    public int deleteAnnouncement(int id) {
+        LOG.debug("delete announcement with id: " + id);
+        final SqlSession session = factory.openSession();
+        int deletedId = -1;
+        try {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            deletedId = mapper.deleteAnnouncement(id);
+            session.commit();
+        } catch (Exception e) {
+            LOG.error(e, "Couldn't delete announcement with id:", id);
```

Comment:
Throw ServiceRuntimeException instead so we know something went wrong. Now it just fails silently.